src/main/java/sirius/db/mixing/types/BaseEntityRef.java

```diff
@@ -302,4 +305,16 @@ public String toString() {
             return "(empty)";
         }
     }
+
+    private E fetchValue(boolean useSecondary) {
+        if (value == null && id != null) {
+            Optional<E> entity = useSecondary ? findInSecondary(type, id) : find(type, id);
+            if (entity.isPresent()) {
+                value = entity.get();
```

Comment:
In my opinion we should not store/cache the value when it was fetched from secondary but just return it. This could pollute a cache with outdated data.